packages/js/product-editor/src/hooks/use-product-attributes.ts

```diff
@@ -69,12 +68,13 @@ export function useProductAttributes( {
 
 	const enhanceAttribute = (
 		globalAttribute: ProductAttribute,
-		terms: ProductAttributeTerm[]
+		allTerms: ProductAttributeTerm[]
 	) => {
 		return {
 			...globalAttribute,
-			terms: terms.length > 0 ? terms : undefined,
-			options: terms.length === 0 ? globalAttribute.options : [],
+			terms: ( allTerms || [] ).filter( ( term ) =>
+				globalAttribute.options.includes( term.name )
+			),
```

Comment:
There was an issue here, where we always passed the `productId` to the `getProductAttributeTerms`, but this meant that any newly added global attributes would return an empty list as they were not saved yet.
I removed the `product` part so it always returns all the `terms` of the specific attribute. This did mean I had to add this `filter` to only add the terms that are part of the product, by comparing it to `options` (the list of term names ).